var/spack/repos/builtin/packages/ghost/package.py

```diff
@@ -0,0 +1,80 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Ghost(CMakePackage):
+    """GHOST: a General, Hybrid and Optimized Sparse Toolkit
+    
+       This library provides highly optimized building blocks for implementing sparse iterative eigenvalue and linear solvers
+       multi- and manycore clusters and on heterogenous CPU/GPU machines. For an iterative solver library using these kernels, see 
+       the phist package.
+    
+    """
+
+    homepage = "https://www.bitbucket.org/essex/ghost/"
+    url      = "https://bitbucket.org/essex/ghost/"
+
+    version('develop', git='https://bitbucket.org/essex/ghost/ghost.git', branch='devel')
+                                
+
+    def cmake_args(self):
+        spec=self.spec
+        cblas_include_dir=''
+        if '~mkl' in spec:
+            cblas_include_dir='-DCBLAS_INCLUDE_DIR='+spec['blas'].prefix.include
+
+        args = ['-DGHOST_ENABLE_MPI:BOOL=%s' % ('ON' if '+mpi' in spec else 'OFF'),
+                '-DGHOST_USE_CUDA:BOOL=%s'   % ('ON' if '+cuda' in spec else 'OFF'),
+                '-DGHOST_USE_SCOTCH:BOOL=%s'   % ('ON' if '+scotch' in spec else 'OFF'),
+                '-DGHOST_USE_ZOLTAN:BOOL=%s'   % ('ON' if '+zoltan' in spec else 'OFF'),
+                '-DBUILD_SHARED_LIBS:BOOL=%s' % ('ON' if '+shared' in spec else 'OFF'),
+                cblas_include_dir,
+                ];
+
+        return args
+
+    def check(self):
+        make('test')
+
+    variant('shared',  default=True, description='Enables the build of shared libraries')
+    variant('mpi', default=True, description='enable/disable MPI')
+    variant('cuda',default=False,description='enable/disable CUDA')
```

Comment:
you may want to derive from `CudaPackage` in addition to `CMakePackage`. That will give you this variant and also `arch=` as well as whole lot of constraints to make things consistent.

Reply:
it would have to be a "conditional" cuda package in that case because cuda is a variant. Would that work somehow?

Reply:
It’s conditional